### PR TITLE
Only apply filters on files from a directory, fixes #32

### LINKED ISSUE #32: Anly apply extension filters to recursive directory traversal, not directly passed files

### PR DESCRIPTION
**Pull request by cpojer**

### CODE REVIEW

File: src/Runner.js [hunk 1]
```diff
@@ -90,11 +93,10 @@ function run(transformFile, paths, options) {
...
   }
 
-  getAllFiles(paths)
-    .then(files => files.filter(
-      name => !extensions || extensions.indexOf(path.extname(name)) != -1
-    ))
-    .then(files => {
+  getAllFiles(
+    paths,
+    name => !extensions || extensions.indexOf(path.extname(name)) != -1
```
Comment (fkling) [inline src/Runner.js]: Thinking about it, this is kind of weird IMO. `js` would also match `jsx` and `json`. I think this should be stricter.

Comment (cpojer) [inline src/Runner.js]: good point.

Comment (cpojer) [inline src/Runner.js]: wait, no, you are not right.

extensions is an array and the value passed to indexOf is the extension name of the file.

Comment (fkling) [inline src/Runner.js]: ooooh, ok then :) (I blame JavaScript)

**Pull request merged**